### PR TITLE
Radioisotope Thermoelectric Generators

### PR DESCRIPTION
**Pull request by Core0verload**
RTGs are simple and reliable nuclear generators. They lack any controls, they start to provide power as soon as they are created, and turn off only when deconstructed. I intend them to be used in lavaland/space ruins instead of magic SMESes.
rtg
Standard RTG outputs 1000w of power, 4000w on T4.
Advanced RTG outputs 2500w of power, 10000w on T4.

Abductors have a pseudo-RTG power source, intended to be used on large abductor ruins. It's called "Void Core", and draws energy out of nowhere, just like that. It outputs 400000w. 
voidcore
When disassembled, it drops T4 parts, alien circuit board and a lesser "void core" - basically an infinite power cell.
voidcell

Token:  #18534

### CODE REVIEW

**Comment by evsey9** [comment]
Can it explode?

**Comment by Core0verload** [comment]
Abductor power cell can, large core can't, RTG can't.

**Comment by evsey9** [comment]
Disappointed 0/10

File: code/modules/power/rtg.dm [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+/obj/machinery/power/rtg/process()
+	..()
+	add_avail(power_gen)
+
+/obj/machinery/power/rtg/RefreshParts()
+	power_gen = initial(power_gen)
+	var/part_level = 0
+	for(var/obj/item/weapon/stock_parts/SP in component_parts)
+		part_level += SP.rating
```
Comment (lordpidey) [inline code/modules/power/rtg.dm]: Isn't there a better way to do this?  I don't remember seeing this in turbines.

File: code/modules/power/rtg.dm [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+	power_gen *= part_level
```
Comment (RemieRichards) [inline code/modules/power/rtg.dm]: ```power_gen = initial(power_gen)*part_level```, and then remove the first set to initial.
no point accessing/setting the var twice.

**Comment by Jazaen** [comment]
Normal RTG won't really explode in any spectacular fashion, mind. The absolute worst it could do would be melting down, unless we assume it's NT/Syndicate fuckery, since I am pretty sure normal RTG won't output nearly this much power. (Well, theoretically the most powerful RTG we have made outputs 3000W, but those actually are small nuclear reactors, and weigh around one tonne)
When it comes to ayytech, you could steal some code from fuel tanks and use that to make the large core explode if shot for those XCOM feels.

**Comment by Armhulen** [comment]
Yes, please make the alien ones explode.
Make sure to place one on the aay ship so the aliens have incentive to keep aubductees asleep

**Comment by WJohn** [comment]
Do these things irradiate you if you stand close to them? I know real life RTGs are well made and can't do that but let's just assume these are super duper shittily done.

**Comment by Core0verload** [comment]
>Well, theoretically the most powerful RTG we have made outputs 3000W, but those actually are small nuclear reactors, and weigh around one tonne

Advanced RTG can control it's reaction, and both use better converters, just like singulo engine converts radiation directly with magic of plasma. Future!

>Do these things irradiate you if you stand close to them? I know real life RTGs are well made and can't do that but let's just assume these are super duper shittily done.

I'll make them irradiate surroundings, but only when panel cover is removed.

**Comment by Core0verload** [comment]
RTGs now irradiate everyone around when their panel is open. You can bucklecuff someone to RTG and open it's panel for loads and loads of fun.

**Comment by Armhulen** [comment]
Is the aubductor core in the flying saucer ship?

**Comment by Core0verload** [comment]
No, these are not in game yet. And I don't know where I put this core. It looks like it would fit in a center of ship, but there is already a surgery table there.

**Pull request merged**